gocourse05/zooobservation/main_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+	"time"
+)
+
+func TestDayCameraSaveToServer(t *testing.T) {
+	t.Run("panic", func(t *testing.T) {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Errorf("Should have panic")
+			}
+		}()
+		var s *string
+		fmt.Println(*s)
+	})
+
+	var err error
+	history := &[]HistoryItem{{
+		time:      time.Now(),
+		direction: right,
+		animalID:  1,
+	}}
+
+	dayCamera := DayCamera{
+		screenshot: "day_screenshot.png",
+	}
+
+	err = dayCamera.SaveToServer(*history)
+	if err != nil {
+		t.Errorf("Error saving history to server with DayCam: %v", err)
+	}
+}
+
+func TestNightCameraSaveToServer(t *testing.T) {
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("The code have panic")
+		}
+	}()
+
+	var err error
+	history := &[]HistoryItem{{
+		time:      time.Now(),
+		direction: right,
+		animalID:  1,
+	}}
+
+	nightCamera := NightCamera{
+		screenshot: "night_screenshot.png",
+	}
+
+	err = nightCamera.SaveToServer(*history)
+	if err != nil {
+		t.Errorf("Error saving history to server with NightCam: %v", err)
+	}
+}
+
+func TestDayCamera_DetectMovement(t *testing.T) {
+	direct := left
+	var history []HistoryItem
+
+	tiger := Animal{
+		id: 1,
+		camera: &DayCamera{
+			screenshot: "day_screenshot.png",
+		},
+		species: "tiger",
+	}
+
+	var err error
+	history, err = tiger.camera.DetectMovement(direct, history, tiger.id)
+	if err != nil {
+		t.Errorf("Error saving history to server with DetectMovement() Tiger: %v", err)
+	} else if history[0].direction != direct || history[0].animalID != tiger.id {
```

Comment:
[`t.Fatalf`](https://pkg.go.dev/testing#T.Fatalf) припиняє виконання цього тесту при фейлі, тому краще використовувати цю функцію для спрощення тесту:
```suggestion
	if err != nil {
		t.Fatalf("Error saving history to server with DetectMovement() Tiger: %v", err)
	}
	if history[0].direction != direct || history[0].animalID != tiger.id {
```

Reply:
Змінити для всіх тестів у файлі.